### PR TITLE
new component for Sneakers route

### PR DESCRIPTION
**Pull request by AlexFlorinBucur**
adding a new component to see sneakers by active filters.

### CODE REVIEW

File: src/Routes/Sneakers/Sneakers.jsx [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
 import SneakerGender from "../../components/Sneaker/SneakerGender";
 import SimpleLine from "../../components/UI/SimpleLine";
+import SneakerFilterActive from "../../components/Sneaker/SneakerFilterActive";
 
 const Sneakers = () => {
   const [sneakersData, setSneakersData] = useState([]);
   const [isLoading, setIsLoading] = useState(false);
   const [error, setError] = useState(null);
+  const [activeFilters, setActiveFilters] = useState({});
+  console.log(activeFilters);
```
Comment (malinmilea) [inline src/Routes/Sneakers/Sneakers.jsx]: please remove all logs, it might create confusion to users

File: src/Routes/Sneakers/Sneakers.jsx [hunk 2]
```diff
@@ -49,15 +55,48 @@ const Sneakers = () => {
...
+          // Verificam daca obiectul valueObj contine cheia "price" si daca pretul se incadreaza in intervalul specificat
+          if ("price" in valueObj) {
+            const priceRange = valueObj.price.split("-").map(Number);
+            const minPrice = priceRange[0];
+            const maxPrice = priceRange[1];
+            const sneakerPrice = sneaker.retailPriceCents;
+
+            if (sneakerPrice < minPrice || sneakerPrice > maxPrice) {
+              return false;
+            }
```
Comment (malinmilea) [inline src/Routes/Sneakers/Sneakers.jsx:84]: good job, the descriptions are helpful for the devs to understand the code

File: src/Routes/Sneakers/Sneakers.jsx [hunk 3]
```diff
@@ -49,15 +55,48 @@ const Sneakers = () => {
...
+          }
+
+          // Returnam true daca obiectul trece toate verificarile, ceea ce inseamna ca indeplineste conditiile din valueObj
+          return true;
         });
 
-      // console.log(transformedData);
       setSneakersData(transformedData);
     } catch (err) {
       setError(err.message);
```
Comment (malinmilea) [inline src/Routes/Sneakers/Sneakers.jsx:101]: please avoid usage of the messages from the server, those are useful only for the devs, but a 404 message is not something descriptive for the user

File: src/Routes/Sneakers/Sneakers.jsx [hunk 4]
```diff
@@ -68,8 +107,21 @@ const Sneakers = () => {
...
+  }, [sneakerFetchHandler, params, location]);
+
+  const deleteQueryHandler = (key) => {
+    const queryValue = query.get(key);
+
+    query.delete(key, queryValue);
+
+    if (query.size === 0) {
+      setActiveFilters({});
+      navigate(``);
```
Comment (malinmilea) [inline src/Routes/Sneakers/Sneakers.jsx:118]: ''

File: src/components/Sneaker/SneakerFilter.jsx [hunk 5]
```diff
@@ -12,10 +12,12 @@ const animationTiming = {
   exit: 100,
 };
 
-const SneakerFilter = ({ sneakersData }) => {
+const SneakerFilter = ({ sneakersData, activeFilters }) => {
   const [keyFilter, setKeyFilter] = useState(false);
   const navigate = useNavigate();
   const location = useLocation();
+  // console.log(location.search);
```
Comment (malinmilea) [inline src/components/Sneaker/SneakerFilter.jsx]: rmv

File: src/components/Sneaker/SneakerFilterActive.jsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+              <li>
+                <Link to={""} className={classes["active-filter-list-item"]}>
+                  <div>
+                    <span className={classes["active-filter-name"]}>
+                      Reset all filters!
+                    </span>
+                    <MdOutlineFilterAltOff />
+                  </div>
+                </Link>
+              </li>
```
Comment (malinmilea) [inline src/components/Sneaker/SneakerFilterActive.jsx:37]: please create a separated component

File: src/components/Sneaker/SneakerList.jsx [hunk 7]
```diff
@@ -21,27 +15,25 @@ const SneakerList = ({ sneakersData }) => {
...
-                      return `${size} `;
-                    })}
+            <li className={classes["sneaker-item"]} key={id}>
+              <Link to={`${id}`}>
+                <p className={classes["brand-name"]}>{brandName}</p>
+                <img src={sneakerImage} alt={name} />
+                <div className={classes["sneaker-name-size"]}>
+                  <p>{name}</p>
+                  <p>
+                    {`Available: `}
```
Comment (malinmilea) [inline src/components/Sneaker/SneakerList.jsx]:
```suggestion
                    {'Available: '}
```

**Review by malinmilea — CHANGES_REQUESTED** [review]

File: src/Routes/Sneakers/Sneakers.jsx (hunk 3 above)
Comment (AlexFlorinBucur) [reply src/Routes/Sneakers/Sneakers.jsx:101]: I implemented this above

File: src/Routes/Sneakers/Sneakers.jsx (hunk 2 above)
Comment (AlexFlorinBucur) [reply src/Routes/Sneakers/Sneakers.jsx:84]: I will change that in future

**Review by malinmilea — APPROVED** [review]

**Pull request merged**